### PR TITLE
ajk/multiscale_bug_fixed

### PR DESCRIPTION
**Pull request by ajkhattak**
bug fixed, looping over cells twice, in computing the subgrid return component.

### CODE REVIEW

**Review by lipnikov — APPROVED** [review]
Minor comments, mainly about Amanzi coding style, see doc/standards/cxx/cppguide.html

File: src/pks/PK_DomainFunctionSubgridReturn.hh [hunk 1]
```diff
@@ -71,10 +71,10 @@ void PK_DomainFunctionSubgridReturn<FunctionBase>::Init(
   field_out_suffix_ = blist.get<std::string>("subgrid field suffix");
 
   // check for prefixing -- mostly used in the multisubgrid models
-  if (blist.isParameter("subgrid domain name"))
-    domain_name_ = blist.get<std::string>("subgrid domain name");
+  if (blist.isParameter("subgrid domain set"))
+    dset_ = blist.get<std::string>("subgrid domain set");
   else
-    domain_name_ = "subgrid";
+    dset_ = "subgrid";
```
[the remark this answers is not in the document]
Comment (ecoon) [reply src/pks/PK_DomainFunctionSubgridReturn.hh:77]: Setting default parameter values is usually done on reading the parameter though.  This should probably just be:

    dset_ = blist.get<std::string>("subgrid domain set", "subgrid");

File: src/pks/PK_DomainFunctionSubgridReturn.hh [hunk 2]
```diff
@@ -53,7 +53,7 @@ class PK_DomainFunctionSubgridReturn : public FunctionBase,
 
  private:
 
-  std::string field_out_suffix_, copy_field_out_key_, domain_name_;
+  std::string field_out_suffix_, copy_field_out_key_, dset_;
```
[the remark this answers is not in the document]
Comment (ajkhattak) [reply src/pks/PK_DomainFunctionSubgridReturn.hh:56]: okay, so I changed the name() method to return dset_ now. I think this is what you suggested, right?

File: src/pks/PK_DomainFunctionSubgridReturn.hh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajkhattak) [reply src/pks/PK_DomainFunctionSubgridReturn.hh]: done

File: src/pks/PK_DomainFunctionSubgridReturn.hh [hunk 3]
```diff
@@ -142,32 +140,34 @@ void PK_DomainFunctionSubgridReturn<FunctionBase>::Compute(double t0, double t1)
...
 	
+      for (int c_sg=0; c_sg!=ncells_sg; ++c_sg) {
 	for (int k=0; k!=nfun; ++k) {
-	  val[k] /= ncells_sg;
+	  val[k] += vec_c[k][c_sg] * alpha[k] ;
 	}
-	value_[*c] = val;
       }
+	
+      for (int k=0; k!=nfun; ++k) {
```
[the remark this answers is not in the document]
Comment (ajkhattak) [reply src/pks/PK_DomainFunctionSubgridReturn.hh:162]: tabs removed.

File: src/pks/PK_DomainFunctionSubgridReturn.hh [hunk 4]
```diff
@@ -142,32 +140,34 @@ void PK_DomainFunctionSubgridReturn<FunctionBase>::Compute(double t0, double t1)
...
+	val[k] /= ncells_sg;
```
[the remark this answers is not in the document]
Comment (ajkhattak) [reply src/pks/PK_DomainFunctionSubgridReturn.hh]: tabs removed

**Pull request merged**